source/extensions/network/dns_resolver/cares/dns_impl.cc

```diff
@@ -204,6 +204,13 @@ void DnsResolverImpl::AddrInfoPendingResolution::onAresGetAddrInfoCallback(
       bool can_process_v6 =
           (!parent_.filter_unroutable_families_ || available_interfaces_.v6_available_);
 
+      int min_ttl = INT_MAX;
+      // Loop through cname and get min_ttl
+      for (const ares_addrinfo_cname* cname = addrinfo->cnames; cname != nullptr;
```

Comment:
AFAIK, there isn't an RFC.
Ideally the TTL of the whole request path should matter and in case the request has come for a cname record the TTL of the cname record should be honoured. The shortest of all guarantees lookup to be most recent and accurate.